src/helics/shared_api_library/MessageFederate.h

```diff
@@ -190,6 +225,148 @@ extern "C"
+    /** get the destination endpoint of a message
+    @param message the message object in question
+    @return a string with the destination endpoint
+    */
+    HELICS_EXPORT const char *helicsMessageGetDestination (helics_message_object message);
+    /** get the original source endpoint of a message, the source may have modified by filters or other actions
+    @param message the message object in question
+    @return a string with the source of a message
+    */
+    HELICS_EXPORT const char *helicsMessageGetOriginalSource (helics_message_object message);
+    /** get the original destination endpoint of a message, the destination may have been modified by filters or other actions
+    @param message the message object in question
+    @return a string with the original destination of a message
+    */
+    HELICS_EXPORT const char *helicsMessageGetOriginalDestination (helics_message_object message);
+    /** get the helics time associated with a message
+    @param message the message object in question
+    @return the time associated with a message
+    */
+    HELICS_EXPORT helics_time helicsMessageGetTime (helics_message_object message);
+    /** get the payload of a message as a string
+    @param message the message object in question
+    @return a string representing the payload of a message
+    */
+    HELICS_EXPORT const char *helicsMessageGetString (helics_message_object message);
+    /** get the messageID of a message
+    @param message the message object in question
+    @return the messageID
+    */
+    HELICS_EXPORT int helicsMessageGetMessageID (helics_message_object message);
+    /** check if a flag is set on a message
+    @param message the message object in question
+    @param flag the flag to check should be between [0,15]
+    @return the flags associated with a message
+    */
+    HELICS_EXPORT helics_bool helicsMessageCheckFlag (helics_message_object message, int flag);
+    /** get the size of the data payload in bytes
+    @param message the message object in question
+    @return the size of the data payload
+    */
+    HELICS_EXPORT int helicsMessageGetRawDataSize (helics_message_object message);
+
+    /** get the raw data for a message object
+    @param message a message object to get the data for
+    @param[out] data the memory location of the data
+    @param maxlen the maximum size of information that data can hold
+    @param[out] actualSize  the actual length of data copied to data
+    @param[in,out] err a pointer to an error object for catching errors
+    */
+    HELICS_EXPORT void helicsMessageGetRawData (helics_message_object message, void *data, int maxlen, int *actualSize, helics_error *err);
+
+    /** get a pointer to the raw data of a message
+    @param message a message object to get the data for
+    @return a pointer to the raw data in memory
+    */
+    HELICS_EXPORT void *helicsMessageGetRawDataPointer (helics_message_object message);
```

Comment:
Should we add a `#define` for this `void *` return type?

Reply:
this function should not be available in the swig generated functions, if it is we need to fix something.    It is really only useful in a C context so we can have access to the underlying data without copying the data.  So in C, I would prefer it remain a void * vs char * or some other name to match C conventions for raw data.  If it were a char * it might be interpreted as a string and that could cause issues.  void * can be cast easily.    Like I said this function doesn't make sense in the language bindings since they wouldn't have direct access to raw data safely.